storage_backends.py

```diff
@@ -43,12 +48,49 @@ def __init__(self, *args, **kwargs):
 
         self.client = dropbox.Dropbox(self.config['access_token'])
 
-    def retrieve(self):
+    def retrieve(self, user):
         _, response = self.client.files_download(self.config['path'])
 
         return todotxtio.from_string(response.content.decode('utf-8'))
 
-    def store(self, todos):
+    def store(self, todos, user):
         data = todotxtio.to_string(todos).encode('utf-8')
 
         self.client.files_upload(data, self.config['path'], mode=dropbox.files.WriteMode('overwrite'), mute=True)
+
+
+def _safe_username(username):
+    return username.replace("/", "_")
+
+
+class WebDav(StorageBackend):
+    def __init__(self, *args, **kwargs):
+        super(WebDav, self).__init__(*args, **kwargs)
+
+    def _make_client(self):
+        if 'webdav_login' in self.config and 'webdav_password' in self.config:
+            options = {'webdav_hostname': self.config['webdav_hostname'],
+                       'webdav_login': self.config['webdav_login'],
+                       'webdav_password': self.config['webdav_password']}
+        else:
+            from flask import request
```

Comment:
Can you put this at the very beginning of this file, even if it's not always used?

Reply:
Addressed in latest commits.